cpp/src/arrow/compute/exec/hash_join_node.cc

```diff
@@ -571,24 +573,82 @@ class HashJoinNode : public ExecNode {
     }
   }
 
-  Status StartProducing() override {
-    START_SPAN(span_, std::string(kind_name()) + ":" + label(),
-               {{"node.label", label()},
-                {"node.detail", ToString()},
-                {"node.kind", kind_name()}});
-    END_SPAN_ON_FUTURE_COMPLETION(span_, finished(), this);
+  std::pair<HashJoinImpl*, std::vector<int>> GetPushdownTarget() {
```

Comment:
Since we operate on these pairs everywhere why not create:
```
struct BloomFilterTarget {
  HashJoinImpl* join_impl;
  std::vector<int> column_map;
};
```

It's also takes a bit of reading to figure out what the purpose of `column_map` is so this could be a place to briefly describe that.

Reply:
I only use the `pair` in one spot as far as I can tell. I just use it so that I can use `std::tie` on whoever calls `GetPushdownTarget`. I did add a big comment though